### DISCUSSION TITLE
How to get Path information in from_fn_with_state_arc

### DISCUSSION THREAD

**Discussion by tingfeng-key** [tokio-rs/axum#1440] [Q&A] [answered]
My URL definition: /test/:account/test_url

Then I want to get the account parameter in from_fn_with_state_arc. How can I do this please?

**Comment by davidpdrsn** [comment, marked as answer]
You can run extractors like this https://docs.rs/axum/0.6.0-rc.2/axum/middleware/fn.from_fn.html#running-extractors

**Reply by tingfeng-key** [reply]
But I need to get State in middleware::from_fn, it doesn't seem to be possible

**Reply by tingfeng-key** [reply]
My needs are like this:
Need to store a `HashMap<username, Data>`, I put it in State
In `/login`, the username is stored in the State, and then the Data is first taken out in the` /:username/*` route. In `/:username/get_info`, 
```rust
async fn get_info(data: Data) {

}
```

**Reply by tingfeng-key** [reply]
I see what you mean, it's resolved. The final implementation is as follows:
```rust
.route_layer(middleware::from_fn_with_state(
            states,
            state::account_middleware,
        ))
```

```rust
pub async fn account_middleware<B>(
    State(states): State<SharedState>,
    Path(account): Path<String>,
    mut req: Request<B>,
    next: Next<B>,
) -> Response {

}
```
